roles/openshift_certificate_expiry/library/openshift_cert_expiry.py

```diff
@@ -0,0 +1,633 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+# pylint: disable=line-too-long,invalid-name
+
+"""For details on this module see DOCUMENTATION (below)"""
+
+# router/registry cert grabbing
+import subprocess
+# etcd config file
+import ConfigParser
+# Expiration parsing
+import datetime
+# File path stuff
+import os
+# Config file parsing
+import yaml
+# Certificate loading
+import OpenSSL.crypto
+
+DOCUMENTATION = '''
```

Comment:
@sdodson thoughts on versioning? I'm not sure 1.3/3.3 is sufficient because we would potentially update mid-stream. Backports would further complicate it.